src/Kpoint.jl

```diff
@@ -4,16 +4,17 @@ More generally, a ``k``-point is a block of the Hamiltonian;
 e.g. collinear spin is treated by doubling the number of ``k``-points.
 """
 struct Kpoint{T <: Real, GT <: AbstractVector{Vec3{Int}}}
-    spin::Int                     # Spin component can be 1 or 2 as index into what is
-    #                             # returned by the `spin_components` function
-    coordinate::Vec3{T}           # Fractional coordinate of k-point
-    G_vectors::GT                 # Wave vectors in integer coordinates (vector of Vec3{Int})
-    #                             # ({G, 1/2 |k+G|^2 ≤ Ecut})
-                                  # This is not assumed to be in any particular order
-    mapping::Vector{Int}          # Index of G_vectors[i] on the FFT grid:
-    #                             # G_vectors(basis)[kpt.mapping[i]] == G_vectors(basis, kpt)[i]
-    mapping_inv::Dict{Int, Int}   # Inverse of `mapping`:
-    #                             # G_vectors(basis)[i] == G_vectors(basis, kpt)[mapping_inv[i]]
+    spin::Int                        # Spin component can be 1 or 2 as index into what is
+    #                                # returned by the `spin_components` function
+    coordinate::Vec3{T}              # Fractional coordinate of k-point
+    G_vectors::GT                    # Wave vectors in integer coordinates (vector of Vec3{Int})
+    #                                # ({G, 1/2 |k+G|^2 ≤ Ecut})
+                                     # This is not assumed to be in any particular order
+    mapping::Vector{Int}             # Index of G_vectors[i] on the FFT grid:
+    #                                # G_vectors(basis)[kpt.mapping[i]] == G_vectors(basis, kpt)[i]
+    mapping_inv::Dict{Int, Int}      # Inverse of `mapping`:
+    #                                # G_vectors(basis)[i] == G_vectors(basis, kpt)[mapping_inv[i]]
+    mapping_gpu::AbstractVector{Int} # GPU copy of `mapping` for fast FFTs
```

Comment:
I would make this a full type variable (similar to `GT`)  and included in the definition of Kpoint.
Also do we need the copy, i.e. can we not just put it always on the gpu ? I think we have a few other places but the FFTs where we need the `mapping`, but these are all not performance critical (e.g. IO), so the extra communication should be fine.